content/news/XXXX-XX-XX-gsoc2025-demucs-to-onnx-dhunstack.md

```diff
@@ -0,0 +1,100 @@
+
+- **Separation Backend:**
+  - PyTorch: Runs separation using the original Demucs model.
+  - ONNX: Runs separation by invoking the ONNX CLI tool via subprocess, matching the workflow of the provided example scripts.
+- **Timing:**
+  - The script records detailed timing for each track, printing and saving the results.
+- **SI-SDR Evaluation:**
+  - Computes the SI-SDR (Scale-Invariant Signal-to-Distortion Ratio) for each separated track using the `torchmetrics` library.
+- **Output:**
+  - Results are saved in timestamped folders, with timing and SI-SDR metrics output to JSON files named according to the backend (e.g., `results_onnx.json`).
+
+### How to Run the Benchmark
+
+1. Place your test tracks in the input directory.
+2. Run the benchmark script:
+   ```sh
+   python benchmark.py --backend onnx --input_dir ./testfiles --output_dir ./outputs
+   ```
+3. Review the JSON results for timing and SI-SDR metrics.
+
+## Export and benchmarks
+
+The quality of the model is expected to be equal or slightly worse when exported to ONNX, due to computational requirements. While there are plenty of ways of measuring the benchmarks models (another blog post incoming), we've chosen to measure our models with `SI-SDR` metric, Scale Invariant Signal To Distortion Ratio, on the MusDB dataset. This is the standard metric on which we measure the performance.
+After export, the model's performance is nearly identical.
+
+
+| Stem         | PyTorch Model (dB) | ONNX Model (C++) (dB) |
+|--------------|--------------------|-----------------------|
+| drums.wav    | 9.53               | 9.53                  |
+| bass.wav     | 13.24              | 13.20                 |
+| other.wav    | 12.61              | 12.64                 |
+| vocals.wav   | 7.42               | 7.41                  |
+| **Overall**  | **10.70**          | **10.69**             |
+
+*Table: SI-SDR (dB) comparison for each stem and overall, using torchmetrics. Results are shown for the native PyTorch model and the exported ONNX model running in C++.*
+
+Now that we have a running platform independent high quality ONNX Demucs model that can utilize hardware acceleration and be deployed with C++, we plan to integrate this into Mixxx DJ for future.
+We've prepared example scripts for running the exported Demucs model, which can be used following the instructions documented in our READMEs.
+
+## Plans for future integration into Mixxx
+
+Here's a list of all the PRs that document our incremental changes towards exporting Demucs:
+
+- [feat] Add ONNX computation path — [mixxxdj/demucs#1](https://github.com/mixxxdj/demucs/pull/1)
+- [feat] Add ONNX exportable STFT implementation — [mixxxdj/demucs#3](https://github.com/mixxxdj/demucs/pull/3)
+- [feat] Add ONNX exportable ISTFT implementation — [mixxxdj/demucs#4](https://github.com/mixxxdj/demucs/pull/4)
+- Add ONNX export instructions — [mixxxdj/demucs#5](https://github.com/mixxxdj/demucs/pull/5)
+
```

Comment:
You should mention upstreaming the changes to https://github.com/adefossez/demucs once you opened the PR

Also mention your upcoming talk at ADC2025